File: unplugged_toaster/day01_containts-duplicate/contains_duplicate.cpp
#include <vector>
#include <unordered_set>

using namespace std; 

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        
        unordered_set<int> mp = {};

        for (const auto& it : nums)
        {
            if (mp.contains(it)) { 
                return true;
            }
            mp.insert(it);
        }

    return false; 
    }
};

File: unplugged_toaster/day01_containts-duplicate/contains_duplicate.rs
use std::collections::HashSet; 

struct Solution {}

impl Solution {

    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        
        let mut hash_set: HashSet<i32> = HashSet::new(); 

        for it in nums { 
            if hash_set.contains(&it) { 
                return true; 
            }
            hash_set.insert(it); 
        }

        return false; 
    }
}

fn main() { 
    let test = vec![1, 2, 3, 1];
    let test2 = vec![1, 2, 3, 4];

    println!("Test 1: {}", Solution::contains_duplicate(test));
    println!("Test 2: {}", Solution::contains_duplicate(test2));
}